### DISCUSSION TITLE
Validating auth token from outside PocketBase?

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#1140] [Q&A]
I am trying to implement a custom SSE use case using PocketBase's auth token for authentication. It looks like EventSource sends cookies, so would I just send the PB auth token as a cookie and then use the PocketBase JS SDK server-side to validate the token? Or is there a better way to do it?

More detail: My Deno process sends stdout/stderr into a `logs.db` SQLite db. I want to implement SSE and an SQLite watcher that can stream log events to the client. I didn't want to clutter up the instance databases with Deno logs, which is why it uses its own. But that leaves me needing to implement my own SSE for that db.

**Comment by ganigeorgiev** [comment]
Yes, you can validate the token by sending `POST /api/collections/users/auth-refresh` (or `POST /api/admins/auth-refresh`) request.

I'm not sure how you'll set an explicit cookie (or other header) with `EventSource` though. Technically the SSE spec doesn't forbid sending headers (it is just a HTTP request), but the default `EventSource` implementation in the browsers for example doesn't support it (see https://github.com/whatwg/html/issues/2177).
Some SSE libraries, eg. https://github.com/EventSource/eventsource, extend the w3c api and allow sending custom headers, including cookies.

**Reply by benallfree** [reply]
`new EventSource(url, { withCredentials: true })` sends cookies, at least on FireFox.If that's not standard, I can just pass the PB cookie token as a querystring param.

How do you authenticate EventSource requests?

**Reply by ganigeorgiev** [reply]
> new EventSource(url, { withCredentials: true }) sends cookies, at least on FireFox.If that's not standard, I can just pass the PB cookie token as a querystring param.

If it works with the cookie, then it should be fine (just make sure that the cookie has the `SameSite=Strict` and `Secure` attributes).

---

> How do you authenticate EventSource requests?

Because in PocketBase it is possible to listen for realtime events as guest (depending on the list and view api rules of the collection), initially we don't authorize the sse request. The authentication happens at a later stage with the additional subscriptions submit request (aka. `POST /api/realtime`).
